### PR TITLE
Create Sales Engine Class

### LINKED ISSUE #17: Create SalesEngine Class
### SalesEngine
```rb
se = SalesEngine.from_csv({
  :items     => "./data/items.csv",
  :merchants => "./data/merchants.csv",
})
```
`:items` generates an instance of `ItemRepository` and `:merchants` returns an instance of `MerchantRepository`

### Methods
- [x] from_csv - creates a new instance of SalesEngine with all the items and merchants from CSV files
- [x] items - returns the list of items loaded from CSV
- [x] merchants - returns the list of merchants loaded from CSV

Effort Point 1

### LINKED ISSUE #48: Fix item_spec file tests and ruby versions for each member
### What:
Tests are failing on `BigDecimal.new`
```sh
1) Item#initialize exists
     Failure/Error: unit_price: BigDecimal.new(10.99,4),
     
     NoMethodError:
       undefined method `new' for BigDecimal:Class
     # ./spec/item_spec.rb:12:in `block (3 levels) in <top (required)>'
.
.
.
```

### Why:
Ruby version on Richard's computer is set to 2.7.2, bundler doesn't work on 2.5.3 even after installing for 2.5.3 but works perfectly find with 2.7.2 or higher.

### .ruby-version
Suggest to add `.ruby-version` file at the root of the project directory and add `2.7.2` to the file to set all environments to this version.

### Steps to fix and to catch up on ruby version

1. Upgrade to ruby version to 2.7.2 by running `rbenv install 2.7.2`
2. Add the `.ruby-version` file as described above with the 2.7.2 version
3. run `bundle` at the root directory of project (This will install all gems for this version)
4. use `BigDecimal()` rather than `BigDecimal.new()` as per the depreciation notes on `BigDecimal`

Effort Points 1

### PR DESCRIPTION
**Pull request by redferret**
Implements Sales Engine Class

Refactors `BigDecimal.new()` to `BigDecimal()`
Adds `.ruby-version` file with 2.7.2 as the project ruby version

closes #17
closes #48

- [x] All tests pass

### CODE REVIEW

**Review by NoahZinter — APPROVED** [review]
Great, that reformatting looks good and removing .new makes the BigDecimal tests pass in ruby 2.7.2 now

**Review by mkrumholz — APPROVED** [review]
Ok, good to go here for the time being.

**Comment by robertguturing** [comment]
3 things to think about when doing code review

1.) Does the code work according to what is asked ⛔️. Or is it broken in some way.
2.)  🗣️  For opinions on the code base. Not necessary to have code merged in.  Have you considered using `find_all` here instead of each?
3.) ⛏️ Hey can we make this enumerable multi-line?

File: lib/sales_engine.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  def self.from_csv(files)
+    instance = SalesEngine.new
+    instance.load_items(files[:items])
+    instance.load_merchants(files[:merchants])
+    instance
+  end
+
+  def load_items(file_name)
+    @items = []
+    CSV.foreach(file_name, headers: true, header_converters: :symbol) do |row|
```
Comment (robertguturing) [inline lib/sales_engine.rb:17]: Cool I like this from a performance perspective instead of using `read`

File: lib/sales_engine.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      @items << Item.new(row)
+    end
+  end
+
+  def load_merchants(file_name)
```
Comment (robertguturing) [inline lib/sales_engine.rb:22]: 🗣️ It looks like `load_items` and `load_merchants` are really similar can we refactor this into one method?

**Pull request merged**